Packages/Tracking/Core/Runtime/Scripts/HandPoses/HandPoseDetector.cs

```diff
@@ -0,0 +1,373 @@
+
+                // Each bone in the finger 
+                for (int boneNum = 0; boneNum < serializedHand.Fingers[fingerNum].bones.Length; boneNum++)
+                {
+                    // Get the same bone for both comparison hand and player hand
+                    Bone activeHandBone = playerHand.Fingers[fingerNum].bones[boneNum];
+                    Bone serializedHandBone = serializedHand.Fingers[fingerNum].bones[boneNum];
+
+                    // Get the user defined rotation threshold for the current bone (threshold is defined in the pose scriptable object)
+                    float jointRotationThreshold = GetBoneRotationThreshold(pose, fingerNum, boneNum);
+
+                    // Get hand rotation for both hands
+                    Quaternion activeBoneRotation = activeHandBone.Rotation;
+                    Quaternion serializedBoneRotation = serializedHandBone.Rotation;
+
+                    // Calculate the vector3 rotation between the current bone and the previous one.
+                    Vector3 activeRotEuler = (Quaternion.Inverse(lastBoneRotation) * activeBoneRotation).eulerAngles;
+                    Vector3 serializedRotEuler = (Quaternion.Inverse(lastSerializedBoneRotation) * serializedBoneRotation).eulerAngles;
+
+                    // Calculate angle difference between the active hand and the serialized hand.
+                    float boneDifference = GetDegreeAngleDifferenceXY(serializedRotEuler, activeRotEuler);
+
+                    lastBoneRotation = activeBoneRotation;
+                    lastSerializedBoneRotation = serializedBoneRotation;
+
+                    // If there is a pose validator present, show the joint colours.
+                    if(_handPoseValidator != null)
+                    {
+                        _handPoseValidator.ShowJointColour(fingerNum, boneNum, boneDifference, jointRotationThreshold);
+                    }
+
+                    // If the pose has been detected, use the Hysteresis value to check if it should be undetected
+                    if (_poseAlreadyDetected)
+                    {
+                        if (boneDifference <= (jointRotationThreshold + _hysteresisThreshold) && boneDifference >= (-jointRotationThreshold - _hysteresisThreshold))
+                        {
+                            numMatchedBones++;
+                        }
+                    }
+                    // Otherwise, check if the difference between current hand and serialized hand is within the threshold.
+                    else
+                    {
+                        if (boneDifference <= jointRotationThreshold && boneDifference >= -jointRotationThreshold)
+                        {
+                            numMatchedBones++;
+                        }
+                    }
+                }
+
+                if(numMatchedBones >= pose.GetNumBonesForMatch(fingerNum))
+                {
+                    ++numMatchedFingers;
+                }
```

Comment:
Similar to above, we could check here if the number of fingers is met and `break` out early